src/main/scripts/run_cdap_e2e_test.py

```diff
@@ -0,0 +1,101 @@
+# Copyright © 2023 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import os
+import subprocess
+import zipfile
+import shutil
+import argparse
+import re
+
+
+def run_shell_command(cmd):
+    process = subprocess.run(cmd.split(" "), stderr=subprocess.PIPE)
+    if process.returncode != 0:
+        print("Process completed with error: ", process.stderr)
+    assert process.returncode == 0
+
+
+def get_sandbox_version(directory_path):
+    pattern = re.compile(r"cdap-sandbox-(\d+\.\d+\.\d+)-SNAPSHOT.zip")
+    # Iterate over files and folders in the directory
+    for entry in os.listdir(directory_path):
+        if pattern.match(entry):
+            version_match = pattern.search(entry)
+            if version_match:
+                version = version_match.group(1)
+                return version
+
+
+# Parse command line optional arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('--testRunner', help='TestRunner class to execute tests')
+args = parser.parse_args()
+
+print("Building CDAP Sandbox")
+os.chdir("./plugin")
+run_shell_command("git submodule update --init --recursive --remote")
+run_shell_command("mvn clean install -DskipTests -P templates,spark-dev")
+my_env = os.environ.copy()
+my_env["MAVEN_OPTS"] = "-Xmx1024m -XX:MaxPermSize=128m"
+my_env["_JAVA_OPTIONS"] = "-Xmx32G"
+# Building the plugins
+os.chdir("..")
+hydrator_repository_url = "https://github.com/cdapio/hydrator-plugins.git"
```

Comment:
If we want to use the same script for release branches, we will need to compute `hydrator-plugins` branch as well from `cdap-build` repo based on cdap branch: https://github.com/cdapio/cdap-build/blob/f785173eb3274b4b9d06e162ce09b9671b2eb091/.gitmodules#L8

Reply:
Added a logic to get the base branch (ex: develop) from e2e.yaml and then based on it we clone cdap-build repo and the look for hydrator-plugins branch in gitmodules. (0bed67d8331a626ce2a755d88c72c08104428228)

e2e.yaml workflow

```
- name: Run all e2e tests
if: github.event_name == 'workflow_dispatch' || github.event_name == 'push' || steps.filter.outputs.e2e-test == 'true'
 run: python3 e2e/src/main/scripts/run_cdap_e2e_test.py --testRunner TestRunnerRequired.java --cdapBranch "${{ github.event.pull_request.base.ref }}"
```

Testing output :

<img width="747" alt="Screenshot 2023-12-21 at 13 04 54" src="https://github.com/cdapio/cdap-e2e-tests/assets/98015926/468191c0-6dd2-42ad-8894-fa8aab4a58b2">